src/compiler/scala/tools/nsc/tasty/bridge/TypeOps.scala

```diff
@@ -283,7 +289,26 @@ trait TypeOps { self: TastyUniverse =>
         case ErasedFunctionType(n) => erasedFnIsUnsupported(formatFnType("=>", isErased = true, n, args))
         case FunctionXXLType(n)     => bigFnIsUnsupported(formatFnType("=>", isErased = false, n, args))
         case _ =>
-          u.appliedType(tycon, args)
+          if (ctx.isJava && tycon.typeSymbol === u.definitions.ArrayClass) {
+            val arg0 = args.head
+            val arg1 =
+              arg0 match {
+                case arg0: u.RefinedType if arg0.parents.exists(_ eq ObjectTpeJava) =>
+                  // option B is to make a new "Mode" in ctx that conditionally disables the ObjectTpeJava substitution,
+                  // e.g. when we read the arguments to Array.
```

Comment:
I tried this out, and because it only should work in the case that its also an intersection type with a type parameter where we want to revert the substitution then we'd need another mode as well to check that were in an intersection type and in array arguments

Reply:
so the changes are not very minimal either way